packages/payment-mop-up-job/src/processors/processor.js

```diff
@@ -18,6 +18,34 @@ const limiter = new Bottleneck({
   maxConcurrent: process.env.CONCURRENCY || CONCURRENCY_DEFAULT
 })
 
+const getPaymentJournalStatusCodeFromGovPayErrorStatusCode = govPayErrorStatusCode => {
+  switch (govPayErrorStatusCode) {
+    case GOVUK_PAY_ERROR_STATUS_CODES.EXPIRED:
+    case GOVUK_PAY_ERROR_STATUS_CODES.NOT_FOUND:
+      return PAYMENT_JOURNAL_STATUS_CODES.Expired
+    case GOVUK_PAY_ERROR_STATUS_CODES.USER_CANCELLED:
+      return PAYMENT_JOURNAL_STATUS_CODES.Cancelled
+    case GOVUK_PAY_ERROR_STATUS_CODES.REJECTED:
+      return PAYMENT_JOURNAL_STATUS_CODES.Failed
+  }
+}
+
+const shouldUpdatePaymentJournal = transaction => {
```

Comment:
I think it'd be worth clarifying the name of this function to say what specific conditions it's for, since there are several other circumstances that we update the payment journal anwyay. At the moment we update the payment journal if it's an error, we update the payment journal if it's a success, and only then do we check `shouldUpdatePaymentJournal`. Threw me off a bit initially.